webui/src/components/schema.jsx

```diff
@@ -15,8 +15,10 @@ export function getSchemaOrderedMajorAndMinorFields(schema) {
     const presentation = schema.getIn(['b2share', 'presentation']);
     const properties = schema.get('properties');
 
-    const majorIDs = presentation ? presentation.get('major') : null;
-    const minorIDs = presentation ?  presentation.get('minor') : null;
+    const majorIDs = presentation && presentation.get('major') ?
+        presentation.get('major').filter(id => properties.get(id)) : null;
```

Comment:
If `presentation` or `presentation.get('major')` are not set, `majorIDs` is `null` This will create an issue during the call `majorIDs.map(...)` on the following lines. Same for `minorIDs`

Reply:
You're right, they can become null, but we check majorIDs and minorIDs before usage with the ternary operator. The mapping is performed only when they're not null.

Reply:
Ah yes thanks for pointing that out, I read the code too fast.